script.js

```diff
@@ -1,12 +1,543 @@
-//You can edit ALL of the code here
-function setup() {
-  const allEpisodes = getAllEpisodes();
-  makePageForEpisodes(allEpisodes);
+// level 500
+function getAllTvShowsApi() {
+  const rootDiv = document.getElementById("root");
+  const mainContainer = document.createElement("div");
+  mainContainer.setAttribute("id", "mainContainer");
+  rootDiv.appendChild(mainContainer);
+  const allTvShowData = getAllShows().sort(compare);
+  tvShowHeaderFooter();
+  allTvShowData.forEach((tvShow) => {
+    mainContainer.appendChild(allTvShowsListings(tvShow));
+  });
 }
 
-function makePageForEpisodes(episodeList) {
+function homeReturnBtn() {
+  //button which returns user home
+  let btn = document.createElement("button");
+  btn.style.display = "none";
+  let navBarContainer = document.getElementById("navBarContainer");
+
+  btn.setAttribute("id", "returnBtn");
+  btn.innerText = "Home";
+  navBarContainer.appendChild(btn);
+
+  btn.addEventListener("click", () => {
+    btn.style.display = "none"; // hides button when click
+    let episodeSelector = document.querySelector("#select"); // hide the tv episodes when the button is pressed
+    episodeSelector.style.display = "none";
+
+    let tvShowSelector = document.querySelector("#select-tv"); // hide the tv episodes when the button is pressed
+    tvShowSelector.style.display = "";
+
+    let episodeContainers = document.getElementsByClassName(
+      "episode-container-class-name"
+    ); // An array of episode container
+    if (episodeContainers) {
+      Array.from(episodeContainers).forEach((episodeContainer) => {
+        episodeContainer.style.display = "none";
+      });
+    }
+
+    let tvShowsInfoContainers = document.getElementsByClassName(
+      "infoContainer"
+    );
+    if (tvShowsInfoContainers) {
+      Array.from(tvShowsInfoContainers).forEach((tvShowsInfoContainer) => {
+        tvShowsInfoContainer.style.display = "";
+      });
+    }
+    updateCounter(getAllShows().length, getAllShows().length); // this updates the counter when the return button is fired
+  });
+}
+
+function tvShowHeaderFooter() {
```

Comment:
It is a good practice to use verbs in function names. They usually represent an action, i.e. `renderHeaderAndFooter`